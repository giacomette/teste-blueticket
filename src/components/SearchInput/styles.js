import styled from 'styled-components';
import TextField from '@material-ui/core/TextField';

export const SearchInputContainer = styled.div`
  max-width: 700px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 60px;
  position: relative;

  .MuiButtonBase-root {
    position: absolute;
    right: 4px;
    top: 4px;
  }

  .MuiInputBase-input {
    border-radius: 10px;
  }
`;

export const InputText = styled(TextField)`
  background: #ffffff;
  border-radius: 30px;
  input {
    padding-right: 55px;
  }

  fieldset {
    border-radius: 30px;
    border-color: #ccc;
  }
`;
