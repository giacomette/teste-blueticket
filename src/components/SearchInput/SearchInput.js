import React from 'react';
import PropTypes from 'prop-types';
import { IconButton } from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';
import CircularProgress from '@material-ui/core/CircularProgress';
import { SearchInputContainer, InputText } from './styles';

function SearchInput({ value, onChange, isLoading, onSearch, suggestions }) {
  return (
    <SearchInputContainer>
      <InputText
        variant="outlined"
        value={value}
        fullWidth
        onChange={e => onChange(e.target.value)}
        onKeyDown={e => e.keyCode === 13 && value && onSearch(value)}
        placeholder="FaÃ§a sua busca"
        inputProps={{
          list: 'searchs'
        }}
      />
      <IconButton
        disabled={isLoading || !value}
        aria-label="search"
        onClick={() => onSearch(value)}
      >
        {isLoading ? <CircularProgress size={25} /> : <SearchIcon />}
      </IconButton>

      <datalist id="searchs">
        {suggestions || [].map(item => <option value={item} />)}
      </datalist>
    </SearchInputContainer>
  );
}

SearchInput.prototype = {
  onChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func,
  value: PropTypes.any.isRequired
};

export default SearchInput;
