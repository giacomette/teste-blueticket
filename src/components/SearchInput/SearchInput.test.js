import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchInput from './SearchInput';

describe('SearchInput', () => {
  it('Deve permitir escrever uma busca', () => {
    const searchValue = 'Rua das andorinhas, 14';
    const component = mount(<SearchInput value={searchValue} />);

    const inputValue = component.find('input');

    expect(inputValue.props().value).toBe(searchValue);
  });

  it('Deve permitir selecionar um histórico de busca', () => {
    const searchValue = 'Rua das andorinhas, 14';
    const component = mount(<SearchInput value={searchValue} />);

    const newSearchValue = 'Avenida Beira mar, 540';

    component.setProps({
      value: newSearchValue
    });

    component.update();

    const inputValue = component.find('input');

    expect(inputValue.props().value).toBe(newSearchValue);
  });

  it('Deve preservar o estado do componente ao clicar para realizar a busca', () => {
    const component = renderer.create(<SearchInput />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Deve apresentar corretamente um historico de busca ja realizado', async () => {
    const searchValue = 'Rua das andorinhas, 14';

    const component = mount(
      <SearchInput suggestions={[searchValue]} value={searchValue} />
    );

    const dataList = component.find('datalist');

    expect(dataList.length).toBe(1);
    expect(dataList.text()).toBe(searchValue);
  });

  it('Não deve permitir realizar uma busca sem informar um valor', () => {
    const searchValue = '';

    const component = mount(<SearchInput value={searchValue} />);

    const button = component.find('button[aria-label="search"]');

    expect(button.props().disabled).toBe(true);
  });
});
