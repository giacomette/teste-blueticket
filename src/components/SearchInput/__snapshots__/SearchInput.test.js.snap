// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Deve preservar o estado do componente ao clicar para realizar a busca 1`] = `
<div
  className="sc-bdVaJa iVwOKK"
>
  <div
    className="MuiFormControl-root MuiTextField-root sc-bwzfXH dBrnQm MuiFormControl-fullWidth"
    onKeyDown={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        list="searchs"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="FaÃ§a sua busca"
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-41 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-42"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    aria-label="search"
    className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
  </button>
  <datalist
    id="searchs"
  />
</div>
`;
